src/Microsoft.PowerShell.Commands.Utility/commands/utility/FormatAndOutput/format-hex/Format-Hex.cs

```diff
@@ -228,78 +237,79 @@ private void ProcessObjectContent(PSObject inputObject)
         {
             Object obj = inputObject.BaseObject;
             byte[] inputBytes = null;
-            if (obj is System.IO.FileSystemInfo)
-            {
-                string[] path = { ((FileSystemInfo)obj).FullName };
-                List<string> pathsToProcess = ResolvePaths(path, true);
-                ProcessPath(pathsToProcess);
-            }
 
-            else if (obj is string)
+            switch (obj)
             {
-                string inputString = obj.ToString();
-                inputBytes = Encoding.GetBytes(inputString);
-            }
+                case System.IO.FileSystemInfo fsi:
+                    string[] path = { fsi.FullName };
+                    List<string> pathsToProcess = ResolvePaths(path, true);
+                    ProcessPath(pathsToProcess);
+                    return;
+                case string str:
+                    inputBytes = Encoding.GetBytes(str);
+                    break;
+                case byte b:
+                    inputBytes = new byte[] { b };
+                    break;
+                case byte[] byteArray:
+                    inputBytes = byteArray;
+                    break;
+                case Int32 iInt32:
+                    inputBytes = BitConverter.GetBytes(iInt32);
+                    break;
+                case Int32[] inputInt32s:
```

Comment:
NIT: variable naming is inconsistent with line 257, perhaps just use `i32` and `i32s`?  Same below for int64

Reply:
Fixed.